modules/st2-elastic-field/directive.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+angular.module('main')
+  .directive('st2ElasticField', [
+    '$window',
+    function (window) {
+      function getDimensions($textarea) {
+        var computedStyle = window.getComputedStyle($textarea[0]),
+            dimensions = {
+              width: window.parseInt(computedStyle.getPropertyValue('width'), 10),
+              height: window.parseInt(computedStyle.getPropertyValue('height'), 10),
+              minHeight: window.parseInt(computedStyle.getPropertyValue('min-height'), 10),
+              maxHeight: window.parseInt(computedStyle.getPropertyValue('max-height'), 10)
+            },
+            lineHeight = window.parseInt(computedStyle.getPropertyValue('line-height'), 10);
+        
+        if (computedStyle.getPropertyValue('box-sizing') === 'border-box') {
+          dimensions.outer = {
+            width:  window.parseInt(computedStyle.getPropertyValue('padding-left'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('padding-right'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-left-width'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-right-width'), 10),
+            height: window.parseInt(computedStyle.getPropertyValue('padding-top'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('padding-bottom'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-top-width'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-bottom-width'), 10)
+          };
```

Comment:
I agree, this part looks pretty monstrous. I think it might be possible to improve it using offset/clientHeight and the already calculated height, but I need to investigate a little into browsers' behavior.